model/project_ref.go

```diff
@@ -404,32 +404,43 @@ func (p *ProjectRef) GetPatchTriggerAlias(aliasName string) (patch.PatchTriggerD
 	return patch.PatchTriggerDefinition{}, false
 }
 
-// MergeWithParserProject looks up the parser project with the given project ref id and modifies
+// MergeWithProjectConfig looks up the project configs with the given project ref id and modifies
 // the project ref scanning for any properties that can be set on both project ref and project parser.
-// Any values that are set at the project parser level will be set on the project ref.
-func (p *ProjectRef) MergeWithParserProject(version string) error {
-	parserProject, err := ParserProjectByVersion(p.Id, version)
+// Any values that are set at the project config level will be set on the project ref IF they are not set on
+// the project ref.
+func (p *ProjectRef) MergeWithProjectConfig(version string) error {
+	projectConfig, err := FindProjectConfigToMerge(p.Id, version)
 	if err != nil {
 		return err
 	}
-	if parserProject != nil {
-		if parserProject.PerfEnabled != nil {
-			p.PerfEnabled = parserProject.PerfEnabled
+	if projectConfig != nil {
+		defer func() {
+			err = recovery.HandlePanicWithError(recover(), err, "project and repo structures do not match")
```

Comment:
copy pasta, should be project ref and config.

I'm surprised that`recursivelySetUndefinedFields` actually works here, because they `do` have different structures, and this function assumes that the structs are the same (because projectRef and repoRef are the same). Also, why are we using reflect but then also assigning workstationConfig and etc explicitly?

Reply:
This works because I am assigning a new ProjectRef struct, and then explicitly setting each config property (workstation config etc) explicitly, and only then letting recursivelySetUndefinedFields do its magic since I am now comparing two project ref structs.